db-service/lib/common/DatabaseService.js

```diff
@@ -111,11 +117,18 @@ class DatabaseService extends cds.Service {
    * @param {string} tenant
    */
   async disconnect(tenant) {
-    const pool = this.pools[tenant]
-    if (!pool) return
-    await pool.drain()
-    await pool.clear()
-    delete this.pools[tenant]
+    const _disconnect = async tenant => {
+      const pool = this.pools[tenant]
+      if (!pool) {
+        return
+      }
+      await pool.drain()
+      await pool.clear()
+      delete this.pools[tenant]
+    }
+    if (tenant == null)
+      return Promise.all(Object.keys(this.pools).map(_disconnect))
+    return _disconnect(tenant)
```

Comment:
```suggestion
      if (tenant == null)
        return Promise.all(Object.keys(this.pools).map(this.disconnect))
      const pool = this.pools[tenant]
      if (!pool) return
      await pool.drain()
      await pool.clear()
      delete this.pools[tenant]
    }
```

Reply:
It might be more understandable, but is also an infinite loop.